indent expects wrong indentation for logical/arithmetic expressions

<!--
This template is for bug reports. If you are reporting a bug, please continue on. If you are here for another reason, please see below:

1. To propose a new rule: http://eslint.org/docs/developer-guide/contributing/new-rules
2. To request a change: http://eslint.org/docs/developer-guide/contributing/changes
3. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**What version of ESLint are you using?**

3.1.1

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```js
indent: [
    'error',
    4,
    {
        SwitchCase: 1
    }
]
```

**What did you do? Please include the actual source code causing the issue.**

```js
(function() {
    var x = 0 ||
        {
            a: 1,
            b: 2
        };

    var y = x ||
        true;
})();
```

**What did you expect to happen?**

I expected it not to have an error about the indentation on lines 3-6

**What actually happened? Please include the actual, raw output from ESLint.**

```sh
/home/user/test.js
  3:13  error  Expected indentation of 4 space characters but found 8   indent
  4:17  error  Expected indentation of 8 space characters but found 12  indent
  5:17  error  Expected indentation of 8 space characters but found 12  indent
  6:13  error  Expected indentation of 4 space characters but found 8   indent
  8:9   error  'y' is defined but never used                            no-unused-vars

✖ 5 problems (5 errors, 0 warnings)
```

--- a/lib/rules/no-restricted-properties.js
+++ b/lib/rules/no-restricted-properties.js
@@ -15,7 +15,7 @@ module.exports = {
     meta: {
         docs: {
             description: "disallow certain properties on certain objects",
-            category: "Node.js and CommonJS",
+            category: "Best Practices",
             recommended: false
         },
 

